packages/bvaughn-architecture-demo/components/sources/utils/getExpressionRange.ts

```diff
@@ -0,0 +1,24 @@
+export default function getExpressionRange(
+  string: string,
+  cursorIndex: number
+): [startIndex: number, endIndex: number] {
+  let startIndex = cursorIndex;
+  while (startIndex >= 0) {
+    const prevCharacter = string.charAt(startIndex - 1);
+    if (prevCharacter === "." || prevCharacter === " " || prevCharacter === "{") {
```

Comment:
I know I have zero context here :sweat_smile:

If it is a space, dot, or backward curly bracket, we want to break out of the loop? I wonder if it makes sense to extract i.e. `"."` into a constant? Or have an object with the characters we don't want to or can't get the expression range from?

I know I've zero context here, feel free to ignore me, just got curious from discord. :smile: